packages/run-protocol/src/feeDistributor.js

```diff
@@ -0,0 +1,302 @@
+      assert.typeof(share, 'bigint', `${kw} must be a bigint; got ${share}`);
+      totalShares += share;
+      return {
+        share,
+        destination,
+      };
+    });
+  return { shares, totalShares };
+};
+
+/** @typedef {ReturnType<typeof makeShareConfig>} ShareConfig */
+
+/**
+ * shareFees tries to be tolerant of single points of failure.
+ *
+ * @param {Payment} payment
+ * @param {Issuer} issuer
+ * @param {ShareConfig} shareConfig
+ */
+export const shareFees = async (payment, issuer, { shares, totalShares }) => {
+  // Divide the payment into shares.
+  /** @type {Amount<'nat'>} */
+  const totalAmount = await E(issuer).getAmountOf(payment);
+  if (AmountMath.isEmpty(totalAmount)) {
+    return;
+  }
+  const { value: totalValue, brand } = totalAmount;
+
+  let remainder = totalValue;
+  /** @type {[ERef<FeeDestination>, Amount][]} */
+  const destinedAmountEntries = shares
+    .map(({ share, destination }, i) => {
+      const value =
+        i === shares.length - 1
+          ? remainder
+          : (totalValue * share) / totalShares;
+      remainder -= value;
+      /** @type {[ERef<FeeDestination>, Amount]} */
+      const entry = [destination, { value, brand }];
+      return entry;
+    })
+    .filter(([_, amt]) => !AmountMath.isEmpty(amt));
+
+  // Split the payment and assert we have a conservation of the total.
+  const sharedPayment = await E(issuer).splitMany(
+    payment,
+    destinedAmountEntries.map(([_dst, amt]) => amt),
+  );
+
+  // Send each nonempty payment to the corresponding destination.
+  await Promise.all(
+    destinedAmountEntries.map(([destination], i) =>
+      E(destination).pushPayment(sharedPayment[i], issuer),
+    ),
+  );
+};
+
+/** @type {ContractStartFn} */
+export const start = async zcf => {
+  const { timerService, collectionInterval, ...rest } = zcf.getTerms();
+
+  /** @type {ShareConfig} */
+  let shareConfig = makeShareConfig();
+
+  /** @type {MapStore<CollectibleContractFacet, FeeCollector>} */
+  const sourceToCollector = makeScalarMapStore();
+  const feeIssuer = await E(E(zcf).getZoeService()).getFeeIssuer();
+
+  const distributeFees = payment => shareFees(payment, feeIssuer, shareConfig);
+
+  const schedulePayments = async () => {
+    if (shareConfig.totalShares <= 0n) {
+      return;
+    }
+    await Promise.all(
+      [...sourceToCollector.values()].map(collector =>
+        E(collector).collectFees().then(distributeFees),
+      ),
+    );
+  };
+
+  const publicFacet = Far('feeDistributor publicFacet', {
+    distributeFees,
+  });
+
+  const zoe = zcf.getZoeService();
+  const creatorFacet = Far('feeDistributor creatorFacet', {
+    addFeeSource: async sourceP => {
+      const [source, collector] = await Promise.all([
+        sourceP,
+        makeFeeCollector(zoe, sourceP),
+      ]);
+      sourceToCollector.init(source, collector);
+      if (shareConfig.totalShares <= 0n) {
+        return;
+      }
+      // Run once immediately for this source.
```

Comment:
why?

Reply:
To both ensure we don't have a bad fee collector, and to catch up in case we're adding a fee collector much later than we started distributing fees.